src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -2046,3 +2044,200 @@ def get_custom_locations_oid(cmd, cl_oid):
         log_string += "Unable to enable the 'custom-locations' feature. " + str(e)
         logger.warning(log_string)
         return ""
+
+
+def troubleshoot(cmd, client, resource_group_name, cluster_name, kube_config=None, kube_context=None, no_wait=False, tags=None):
+
+    try:
+
+        # C:\Diagnoser\azure-cli-extensions\src\connectedk8s\azext_connectedk8s\custom.py
+        logger.warning("Diagnoser running. This may take a while ...\n")
+        absolute_path=os.path.abspath(os.path.dirname(__file__))
+
+        storage_space_available = True
+        # Setting default values for all checks as True
+        diagnostic_checks = {
+        "msi_cert_expiry_check" :True,
+        "kap_security_policy_check" : True,
+        "kap_cert_check" : True,
+        "diagnoser_check" : True,
+        "msi_cert_check" : True,
+        "agent_version_check" : True,
+        "arc_agent_state_check" : True
+        }
+
+        # Setting kube_config
+        kube_config = set_kube_config(kube_config)
+
+        # Loading the kubeconfig file in kubernetes client configuration
+        load_kube_config(kube_config, kube_context)
+        configuration = kube_client.Configuration()
+
+        # Install helm client
+        helm_client_location = install_helm_client()
+
+        # Checking the connection to kubernetes cluster.
+        # This check was added to avoid large timeouts when connecting to AAD Enabled AKS clusters
+        # if the user had not logged in.
+        check_kube_connection(configuration)
+        utils.try_list_node_fix()
+
+        # Fetch Connected Cluster for agent version
+        connected_cluster = get_connectedk8s(cmd, client, resource_group_name, cluster_name)
```

Comment:
There could be one error scenario where customer might pass an invalid/different cluster name from the one present in the config map. I think, it would be better to check if this input(rg_name, cc_name) matches with the config map or not and display a warning/error saying the input is invalid. Some check depend on cc response and others depend on arc-agents. so there should be 1:1 mapping between the data at both places, otherwise the troubleshoot results become invalid

Reply:
Thanks for spotting this out. Made the required changes.